addons/mail/static/src/core/messaging_service.js

```diff
@@ -205,6 +205,45 @@ export class Messaging {
                         );
                     }
                     break;
+                case "discuss.channel/delete":
+                    {
+                        const thread = this.threadService.insert({
+                            id: notif.payload.id,
+                            model: "discuss.channel",
+                        });
+                        const filteredStarredMessages = [];
+                        let starredCounter = 0;
+                        for (const msg of this.store.discuss.starred.messages) {
+                            if (msg.resModel !== thread.model || msg.resId !== thread.id) {
+                                filteredStarredMessages.push(msg);
+                            } else {
+                                starredCounter++;
+                            }
+                        }
+                        this.store.discuss.starred.messages = filteredStarredMessages;
+                        this.store.discuss.starred.counter -= starredCounter;
+                        this.store.discuss.inbox.messages =
+                            this.store.discuss.inbox.messages.filter(
+                                (msg) => msg.resModel !== thread.model || msg.resId !== thread.id
+                            );
+                        this.store.discuss.inbox.counter -= thread.message_needaction_counter;
+                        this.store.discuss.history.messages =
+                            this.store.discuss.history.messages.filter(
+                                (msg) => msg.resModel !== thread.model || msg.resId !== thread.id
+                            );
+                        for (const message of thread.messages) {
+                            delete this.store.messages[message.id];
+                        }
+                        this.threadService.remove(thread);
+                        if (thread.localId === this.store.discuss.threadLocalId) {
+                            this.threadService.setDiscussThread(
+                                this.store.discuss.channels.threads[0]
+                                    ? this.store.threads[this.store.discuss.channels.threads[0]]
+                                    : this.store.discuss.inbox
```

Comment:
I don't like the idea to reset to another channel, I think it should either set an empty thread (or fallback to inbox possibly).
Going involuntarily to another channel could unexpectedly mark its message as read, among other things.

Reply:
Okay then we set it to Inbox.